src/main/java/org/terasology/lost/OnSpawnSystem.java

```diff
@@ -15,26 +15,68 @@
  */
 package org.terasology.lost;
 
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityBuilder;
 import org.terasology.entitySystem.entity.EntityManager;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
 import org.terasology.logic.inventory.InventoryComponent;
 import org.terasology.logic.inventory.InventoryManager;
 import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.math.Side;
 import org.terasology.registry.In;
+import org.terasology.logic.console.Console;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.world.WorldProvider;
+import org.terasology.structureTemplates.events.SpawnStructureEvent;
+import org.terasology.structureTemplates.util.BlockRegionTransform;
+
+import static org.terasology.lost.LevelSpawnSystem.getGroundHeight;
 
 @RegisterSystem(RegisterMode.AUTHORITY)
 public class OnSpawnSystem extends BaseComponentSystem {
     @In
     private EntityManager entityManager;
     @In
     private InventoryManager inventoryManager;
+    @In
+    private Console console;
+    @In
+    private AssetManager assetManager;
+    @In
+    private WorldProvider worldProvider;
 
     @ReceiveEvent
     public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, InventoryComponent inventory) {
         inventoryManager.giveItem(player, null, entityManager.create("Lost:antrumSabre"));
+        ProgressTrackingComponent progressTrackingComponent = new ProgressTrackingComponent();
+        progressTrackingComponent.addLevel("Lost:well", "Beach", "Coast", "Lakeshore");
+        progressTrackingComponent.addLevel("Lost:templePlasmaOfFire", "Tropical seasonal forest", "Tropical rain " +
+                "forest", "Temperate deciduous forest", "Temperate rain forest");
+        progressTrackingComponent.addLevel("Lost:stonehengeWipedOutKey", "Marsh", "Shrubland", "Grassland");
+        progressTrackingComponent.addLevel("Lost:pyramidBladesOfTruth", "Tundra", "Bare", "Scorched", "Temperate " +
+                "desert", "Subtropical desert");
+        player.addComponent(progressTrackingComponent);
+
+        LocationComponent loc = player.getComponent(LocationComponent.class);
+        Vector3f playerLocation = loc.getWorldPosition();
+        Prefab prefab =
```

Comment:
For lines like this where you're using something nice like `orElse(null)` would it make sense to then also harden for the case where you _do_ get a null, to provide some sort of friendly error message and an explicit runtime error rather than risk hitting an NPE?

More of a question / recommendation than necessarily something to do everywhere. Just seems like a good practice :-)